book/src/interface_builder.md

```diff
@@ -0,0 +1,135 @@
+# Interface Builder
+
+Until now, whenever we constructed pre-defined widgets we relied on the [builder pattern](https://doc.rust-lang.org/1.0.0/style/ownership/builders.html).
+This way we ended up with the most recent form of our trusty "Hello World!" app.
+
+<span class="filename">Filename: listings/widgets/2/main.rs</span>
+```rust,no_run
+{{#rustdoc_include ../listings/widgets/2/main.rs}}
+```
+
+Creating widgets directly from code is perfectly fine.
+However, with most toolkits you can describe your user interface with a markup language and GTK is no exception here.
+For example the following `xml` snippet describes the window widget of the "Hello World!" app. 
+
+<span class="filename">Filename: listings/interface_builder/1/window.ui</span>
+```xml
+{{#rustdoc_include ../listings/interface_builder/1/window.ui}}
+```
+
+The most outer tag always has to be the `<interface>`.
+Then you start listing the elements you want to describe.
+In our case, we want to have a `gtk::ApplicationWindow`.
+These `xml` files are independent of the programming language, which is why the classes have the original names.
+Luckily, they all convert like this: `gtk::ApplicationWindow` → `GtkApplicationWindow`.
+We want to access the window later on, so we also give it an `id`.
+Then we can specify properties which are specified [here](https://docs.gtk.org/gtk4/class.ApplicationWindow.html) for `ApplicationWindow`.
+Since `ApplicationWindow` can contain other widgets we use the `<child>` tag to add a `Button`.
+
+To instantiate the widgets described by the `xml` files we use [`gtk::Builder`](../docs/gtk4/struct.Builder.html)[^1].
+
+<span class="filename">Filename: listings/interface_builder/1/main.rs</span>
+```rust,no_run
+{{#rustdoc_include ../listings/interface_builder/1/main.rs:build_ui}}
+```
+
+This is a bit disappointing.
+Even though we have already described the UI in the markup file, the amount of code is still pretty much the same.
+There are still cases where it is valuable to know of the existence of `gtk::Builder`.
+We will see for example that [`ShortcutsWindow`](../docs/gtk4/struct.ShortcutsWindow.html) is quite a bit easier to instantiate that way.
```

Comment:
Not really because gtk provides a pretty handy way to do that. In general, you very often don't want GtkBuilder directly

Reply:
You mean the trick with GResources?

Reply:
That *trick* is explained in GtkApplication documentation https://docs.gtk.org/gtk4/class.Application.html#automatic-resources, since the docs of the Application class should be the most visited/read ones I would refrain from calling it a trick.

Reply:
No reason to get worked up about it.
If that's what Bilal meant, I can't use it anyways because Ressources will be only introduced later.